### PR TITLE
Update _functions.php

### PR DESCRIPTION
**Pull request by deepend-tildeclub**
you get these two errors on the page it takes you when you sign out of the website.

Warning: preg_replace(): Compilation failed: unrecognized character follows \ at offset 3 in /home/retrodig/public_html/zarksfallenangels/_functions.php on line 131

Deprecated: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in /home/retrodig/public_html/zarksfallenangels/_functions.php on line 133

### CODE REVIEW

**Comment by RedDragonWebDesign** [comment]
Clicking "Sign Out" in the top right logs me out and quickly redirects me to the homepage. In the blink of an eye. If there were a warning, I would think it would print it, which would mess up the header that does the redirect, and I'd be able to see it.

Any hint for steps to reproduce?

**Comment by RedDragonWebDesign** [comment]
I found this error. It's not in the logout system, it's when viewing a forum thread.

http://localhost/src/forum/viewtopic.php?tID=1

File: src/_functions.php [hunk 1]
```diff
@@ -118,20 +118,20 @@ function parseBBCode($strText) {
...
+	$strText = preg_replace("/\\[url=(.*?)\\](.*?)\\[\\/url\\]/i", "<a href='$1' target='_blank'>$2</a>", $strText); // Links with Titles
 
-	$strText = preg_replace("/\[color=(.*)\](.*)\[\/color\]/i", "<span style='color: $1'>$2</span>", $strText); // Text Color
+	$strText = preg_replace("/\\[color=(.*?)\\](.*?)\\[\\/color\\]/i", "<span style='color: $1'>$2</span>", $strText); // Text Color
 
 	$strText = str_replace("[/youtube]", "[/youtube]\n", $strText);
-	$strText = preg_replace("/\[youtube\](http|https)(\:\/\/www\.youtube\.com\/watch\?v\=)(.*)\[\/youtube\]/i", "<iframe class='youtubeEmbed' src='http://www.youtube.com/embed/$3?wmode=opaque' frameborder='0' allowfullscreen></iframe>", $strText);
-	$strText = preg_replace("/\[\youtube\](http|https)(\:\/\/youtu\.be\/)(.*)\[\/youtube\]/i", "<iframe class='youtubeEmbed' src='http://www.youtube.com/embed/$3?wmode=opaque' frameborder='0' allowfullscreen></iframe>", $strText);
+	$strText = preg_replace("/\\[youtube\\](http|https)(:\\/\\/www\\.youtube\\.com\\/watch\\?v=)(.*?)\\[\\/youtube\\]/i", "<iframe class='youtubeEmbed' src='http://www.youtube.com/embed/$3?wmode=opaque' frameborder='0' allowfullscreen></iframe>", $strText);
```
Comment (RedDragonWebDesign) [inline src/_functions.php:130]: OK, so the reason this line was originally throwing an error can be demonstrated here:

https://regex101.com/r/h9famE/3

Old versions of PHP were running regex version `PCRE (PHP <7.3)`. You'll notice that `PCRE (PHP <7.3)` is selected in the flavor section, and there are no errors (no red spots in the regular expression.

image

But PHP 8.1 runs `PCRE2 (PHP >=7.3)`. If you click on flavor `PCRE2 (PHP >=7.3)`, you'll notice we immediately get a red spot (an error).

image

The fix is simply to delete the `\` in front of the word `youtube`.

It looks like your changes break the code even more without fixing it. It becomes a little more obvious at this link which uses your changed regex, which has 4 red spots:

https://regex101.com/r/Zc53Rm/1

image

Anyway, with this pointed out and with this regex101 tool, think you can update this patch to just remove the appropriate backslashes and change nothing else? I think that's the way to go.

**Review by RedDragonWebDesign — CHANGES_REQUESTED** [review]

**Comment by deepend-tildeclub** [comment]
> I found this error. It's not in the logout system, it's when viewing a forum thread.
> 
> http://localhost/src/forum/viewtopic.php?tID=1

Seems it is behaving slightly different then it was earlier today.  I put the _functions.php file back to master and it gives me the errors on the main page logged in or not.   Also also see the errors in the forum thread as you do.

File: src/_functions.php (hunk 1 above)
Comment (deepend-tildeclub) [reply src/_functions.php:130]: > Anyway, with this pointed out and with this regex101 tool, think you can update this patch to just remove the appropriate backslashes and change nothing else? I think that's the way to go.

Thanks.. will look closer at that.. also thanks for the website.  Never heard of regex101.com.  Looks very useful :)

**Comment by RedDragonWebDesign** [comment]
Alright, I just went ahead and merged a patch that deleted that one backslash from `\youtube` that I mentioned above, because it was causing the only remaining warnings on the forum viewtopic page.

Can you `pull` and `checkout` the latest master branch and see if this patch is still needed? I think we might be able to abandon this patch, unless it fixes something else.

**Pull request closed without merging**